Tools/TestWebKitAPI/Tests/WTF/WeakPtr.cpp

```diff
@@ -1201,6 +1201,23 @@ TEST(WTF_WeakPtr, WeakHashMapConstObjects)
     }
 }
 
+TEST(WTF_WeakPtr, WeakHashMapRemoveIterator)
+{
+    WeakHashMap<Base, int> weakHashMap;
+    Vector<std::unique_ptr<Base>> objects;
+    for (unsigned i = 0; i < 13; ++i) {
+        auto object = makeUnique<Base>();
+        weakHashMap.add(*object, 0);
+        objects.append(WTFMove(object));
+    }
+    while (!objects.isEmpty()) {
+        auto it = weakHashMap.find(*objects.last());
```

Comment:
Shouldn't we be removing the first here or removing the last item below??

Reply:
This reproduces the crash reliably so I don’t think it needs changing. The idea is to destroy an object, so that there is a null reference in the weak hash, then we try to delete an iterator to another object in the hash map. The iterator sometimes would get invalidated by the removal of the null reference.